src/hera/workflows/script.py

```diff
@@ -24,6 +25,12 @@
     cast,
 )
 
+if sys.version_info >= (3, 9):
+    from typing import Annotated
+else:
+    from typing_extensions import Annotated
+
+
 from typing_extensions import ParamSpec, get_args, get_origin
```

Comment:
Not strong opinion. looks great if we update this statement as well.

Reply:
👍  Will leave to a separate PR to avoid risking a behaviour change